### PR TITLE
add gtihub workflow to deploy on ropsten testnet and  run integration test

### LINKED ISSUE #183: Deploy to Testnet
Tag a version of stable NF3 and deploy to ropsten testnet

### PR DESCRIPTION
**Pull request by LijuJoseJJ**
This PR add a github action workflow which will test all PR to master branch by deploying code to infura testnet(ropsten) and run integration test (test/http.mjs and test/neg-http.mjs).

fixes #183 

this PR does few changes to make continues integration works (deploy at ropsten network).
1. lowers `block_stake` and `bond`,  `1 eth` and `10 eth` to `1 wei` and `10 wei` respectively.
2. added basic auth logic to web3 connection by passing `INFURA_PROJECT_SECET` env (this enable a basic security feature as infura set transaction limit per day)

NOTE: Infura testnet deployment `./start-nightfall -t` cannot be tested locally (unless fews changes are made) , because it has dependency on fews secret environment variables
1.  `ETH_PRIVATE_KEY`
2. `INFURA_PROJECT_ID`
3. `INFURA_PROJECT_SECRET`

changes to make Infura testnet deployment and then integration test to work:-
1. update FROM_ADDRESS in `./docker-compose.testnet.yml` and in `./package.json` with your own EOA for which have balance in rospten network.
2. set environment variable `ETH_PRIVATE_KEY` to your EOA's private key.
3 set env `INFURA_PROJECT_ID` and `INFURA_PROJECT_SECRET` values from your own infura project or use one which github action workflow is using (slack me for the these values :) )

also this PR fixes github action workflow which run `npm run test-gas` by updating `docker-compose.yml` to expect value for TRANSACTIONS_PER_BLOCK from environment variable, if not set take default value "2".

### CODE REVIEW

**Comment by ChaitanyaKonda** [comment]
Can we make this ropsten testnet please?

**Comment by LijuJoseJJ** [comment]
> Can we make this ropsten testnet please?

done

File: test/neg-http.mjs [hunk 1]
```diff
@@ -124,102 +138,106 @@ describe('Testing the challenge http API', () => {
...
-          // When a challenge succeeds, the challenger is removed. We are adding them back for subsequent for challenges
-          const result = await chai
-            .request(optimistUrl)
-            .post('/proposer/register')
-            .send({ address: myAddress });
-          txToSign = result.body.txDataToSign;
-          await submitTransaction(txToSign, privateKey, proposersAddress, gas, bond);
-          // console.log('tx hash of challenge block is', txReceipt.transactionHash);
-        } else throw new Error(`Unhandled transaction type: ${type}`);
+        try {
```
Comment (IlyasRidhuan) [inline test/neg-http.mjs:141]: Why do we need `try catch` around the websocket code?

File: test/neg-http.mjs [hunk 2]
```diff
@@ -556,13 +574,19 @@ describe('Testing the challenge http API', () => {
     // if the queue is still running, let's close down after it ends
     // if it's empty, close down immediately
     if (txQueue.length === 0) {
+      console.log('neg-http.mjs after - test suits - if case');
```
Comment (IlyasRidhuan) [inline test/neg-http.mjs:577]: can we remove some of the debug `console.log` in tests. It will de-clutter the output

Comment (LijuJoseJJ) [reply test/neg-http.mjs]: removed

File: test/neg-http.mjs (hunk 1 above)
Comment (LijuJoseJJ) [reply test/neg-http.mjs:141]: whenever error occurs in websocket 'onmessage' emitHandler it does not terminates test suite( throw does not propagate to integration test execution). instead next dependent test check fails.
I added try - catch with console.log to easy debug at such cases

**Review by IlyasRidhuan — APPROVED** [review]

**Pull request merged**